eng/common/pipelines/templates/archetype-typespec-emitter.yml

```diff
@@ -189,26 +193,19 @@ extends:
                 ServiceEndpointUrl: https://api.esrp.microsoft.com
                 MainPublisher: ESRPRELPACMANTEST
 
-          - task: PowerShell@2
-            displayName: Create emitter-package.json
-            inputs:
-              pwsh: true
-              filePath: ./eng/common/scripts/typespec/New-EmitterPackageJson.ps1
-              arguments: >
-                -PackageJsonPath '$(buildArtifactsPath)/lock-files/package.json'
-                -OverridesPath '$(buildArtifactsPath)/overrides.json'
-                -OutputDirectory '$(Build.ArtifactStagingDirectory)'
-              workingDirectory: $(Build.SourcesDirectory)
+          - pwsh: |
+              npm install -g @azure-tools/tsp-client@latest
+            displayName: Install tsp-client
 
-          - task: PowerShell@2
-            displayName: Create emitter-package-lock.json
-            inputs:
-              pwsh: true
-              filePath: ./eng/common/scripts/typespec/New-EmitterPackageLock.ps1
-              arguments: >
-                -EmitterPackageJsonPath '$(Build.ArtifactStagingDirectory)/emitter-package.json'
-                -OutputDirectory '$(Build.ArtifactStagingDirectory)'
-              workingDirectory: $(Build.SourcesDirectory)
+          - pwsh: |
+              tsp-client generate-config-files '$(buildArtifactsPath)/lock-files/package.json' --output-dir '$(Build.ArtifactStagingDirectory)' 
+              --emitter-package-json-path  ${{ parameters.EmitterPackageJsonPath }}
+              --overrides '$(buildArtifactsPath)/overrides.json'
+            displayName: Generate emitter-package.json file
+
+          - pwsh: |
+              tsp-client generate-lock-file --output-dir '$(Build.ArtifactStagingDirectory)' --emitter-package-json-path  ${{ parameters.EmitterPackageJsonPath }}
+            displayName: Generate emitter-package-lock.json file  
```

Comment:
no need for this call now.   `tsp-client generate-config-files` makes an implicit, chained call to `generate-lock-file`